### PR TITLE
[yaml] disable javascript mapping for python >=3.12

### PR DESCRIPTION
**Pull request by Polber**
js2py is not supported in Python 3.12, so to prevent future issues arising from Beam Python 3.12 support, this js2py import is being excluded for Beam installs starting with Python 3.12 and the feature will throw error if the package does not exist.

Tracked in https://github.com/apache/beam/issues/30829

js2py issues
* https://github.com/PiotrDabkowski/Js2Py/issues/282#issuecomment-1327954348
* https://github.com/PiotrDabkowski/Js2Py/issues/317

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Polber** [comment]
R: @tvalentyn

**Comment by Polber** [comment]
R: @robertwb

File: sdks/python/setup.py [hunk 1]
```diff
@@ -368,7 +372,6 @@ def get_portability_package_data():
           'grpcio>=1.33.1,!=1.48.0,<2',
           'hdfs>=2.1.0,<3.0.0',
           'httplib2>=0.8,<0.23.0',
-          'js2py>=0.74,<1',
```
Comment (tvalentyn) [inline sdks/python/setup.py:371]: we can simplify it via an environmental marker: https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#platform-specific-dependencies

Comment (tvalentyn) [reply sdks/python/setup.py:371]: Also let's reference the issue in JS2Py's tracker

**Review by tvalentyn — CHANGES_REQUESTED** [review]
Thank you!

File: sdks/python/setup.py (hunk 1 above)
Comment (Polber) [reply sdks/python/setup.py:371]: Thanks for the heads up - refactored to use environmental marker.

File: sdks/python/setup.py [hunk 2]
```diff
@@ -368,7 +368,7 @@ def get_portability_package_data():
...
+          'js2py>=0.74,<1; python_version<"3.12"',
```
Comment (tvalentyn) [inline sdks/python/setup.py:371]:
```suggestion
          # https://github.com/PiotrDabkowski/Js2Py/issues/317
          'js2py>=0.74,<1; python_version<"3.12"',
```

Comment (Polber) [reply sdks/python/setup.py:372]: Oh sorry I totally misunderstood what you were asking, thanks for adding that!

**Review by tvalentyn — APPROVED** [review]

**Pull request merged**